pkg/apis/imageregistry/v1/types.go

```diff
@@ -255,47 +289,54 @@ type ImageRegistryConfigStorage struct {
 	// for production use. When the pod is removed from a node for any reason,
 	// the data in the emptyDir is deleted forever.
 	// This configuration is EXPERIMENTAL and is subject to change without notice.
+	// +optional
```

Comment:
We need to add a discriminator field and type, i.e.

```
type ImageRegistryConfigStorageType string

const (
    AzureImageRegistryStorageType ImageRegistryConfigStorageType = "azure"
    GCSImageRegistryStorageType ImageRegistryConfigStorageType = "gcs"
    S3ImageRegistryStorageType ImageRegistryConfigStorageType = "s3"
...
)
```

And in the storage struct:
```
    Type ImageRegistryConfigStorageType `json:type`
```
See https://github.com/openshift/api/pull/300

Reply:
Apparently we also need to change our logic to use this field. And we should do it backward compatible. Should it be a separate PR?

Reply:
@adambkaplan ?